replica/mutation_dump.cc

```diff
@@ -405,6 +405,20 @@ future<mutation_reader> make_partition_mutation_dump_reader(
         tracing::trace_state_ptr ts,
         db::timeout_clock::time_point timeout) {
     const auto& tbl = db.local().find_column_family(underlying_schema);
+
+    // We can get a request for a token we don't own.
+    // Just return empty reader in this case, otherwise we will hit
+    // std::terminate because the replica side does not handle requests for
+    // un-owned tokens.
+    {
+        auto erm = tbl.get_effective_replication_map();
+        auto& topo = erm->get_topology();
+        const auto endpoints = erm->get_endpoints_for_reading(dk.token());
+        if (std::ranges::find(endpoints, topo.this_node()->endpoint()) == endpoints.end()) {
+            co_return make_empty_flat_reader_v2(output_schema, std::move(permit));
+        }
+    }
```

Comment:
The check is only valid for the duration it is held. If we lose the range immediately after this }, we'll crash exactly the same.

We need to extend erm's lifetime over the whole query.

IMO the best approach is to accept it in make_partition_mutation_dump_reader() and pass it to mutation_dump_reader's constructor. This shows that we're locking the whole thing against tablet migrations.

Note: most likely, erm is in fact held for the duration of the query, but it's impossible to verify it.

Reply:
`erm` is held up in the `cql3` statement's `cql3::statements::mutation_fragments_select_statement::do_execute()`.

Reply:
Then why not pass it down? So we can use local reasoning rather than non-local reasoning.

Reply:
Done in v2.